### PR TITLE
New package: ddgtk-0.1

### PR DESCRIPTION
**Pull request by reback00**
[ddgtk](https://github.com/gort818/ddgtk) is a GUI version of the command line `dd`. It works similar to [unetbootin](https://unetbootin.github.io/) or [Etcher](https://www.balena.io/etcher/). It makes life so much easier!

I use this tool after Etcher started showing annoying ads and stuff.
Hope it is useful.

### CODE REVIEW

**Comment by reback00** [comment]
Having some errors with ARM builds. Anybody has any idea to how I can solve it?

```
...
=> ddgtk-0.1_1: running pre-configure hook: 02-script-wrapper ...
=> ddgtk-0.1_1: running do_configure ...
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/mesonbuild/mesonmain.py", line 122, in run
    return options.run_func(options)
...
  File "/usr/lib/python3.6/subprocess.py", line 1364, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'glib-compile-resources': 'glib-compile-resources'
The Meson build system
Version: 0.50.1
...
Found pkg-config: /usr/bin/pkg-config (0.29.2)
WARNING: Could not detect glib version, assuming 2.54. You may get build errors if your glib is older.
=> ERROR: ddgtk-0.1_1: do_configure: '${meson_cmd} --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/libexec --bindir=/usr/bin --sbindir=/usr/bin --includedir=/usr/include --datadir=/usr/share --mandir=/usr/share/man --infodir=/usr/share/info --localedir=/usr/share/locale --sysconfdir=/etc --localstatedir=/var --sharedstatedir=/var/lib --buildtype=plain --auto-features=enabled --wrap-mode=nodownload -Db_lto=true -Db_ndebug=true -Db_staticpic=true ${configure_args} . ${meson_builddir}' exited with 2
=> ERROR:   in do_configure() at common/build-style/meson.sh:108
The command "if [ -z "$XLINT" ]; then docker exec -t void hostrepo/common/travis/build.sh $BOOTSTRAP $ARCH; fi" exited with 1.
...
```
from [https://travis-ci.org/void-linux/void-packages/jobs/546344997](https://travis-ci.org/void-linux/void-packages/jobs/546344997)

**Comment by Hoshpak** [comment]
Putting `glib-devel` in `hostmakedepends` should solve it.

**Comment by reback00** [comment]
@Hoshpak 
Thanks a lot man. It worked!

One thing is that I listed `glibc-devel` under `makedepends` based on this text on the [Manual.md](https://github.com/void-linux/void-packages/blob/master/Manual.md)

> ...Typically the `makedepends` will be the only one of the three types of `depends` to include `-devel` packages, and typically only `-devel` packages. ...

I wonder why the instructions in the manual does not work.

**Comment by newbluemoon** [comment]
While building it needs to run `/usr/bin/glib-compile-resources` from `glib-devel`. However, the `glib-devel` under `makedepends` is the the target version. So e.g. the x86_64 host cannot execute an armv7l target binary. That’s why you also have to add it to `hostmakedepends`. But in case the source code also includes headers from `glib-devel` you have to be careful that the ones from the target version are used (there might be arch specific differences). I’m not that familiar with the meson build-style, but think it should take care of this. However, it might be a problem with other styles in general.

**Comment by Hoshpak** [comment]
`glib-devel` is an exception. Most `-devel` packages contain only header files and similar (pkg-config, cmake, etc.) however `glib-devel` also contains binaries to be executed on the host. So while most `-devel` packages should only be used in `makedepends`, putting `glib-devel` in hostmakedepends is perfectly fine when needed.

File: srcpkgs/ddgtk/template [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+build_style=meson
+hostmakedepends="pkg-config glib glib-devel"
+depends="python3 python3-gobject gtk+3"
+short_desc="GUI frontend for dd to create bootable ISO images for Linux"
+maintainer="reback00 <reback00@protonmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/gort818/ddgtk"
+distfiles="https://github.com/gort818/ddgtk/archive/${version}.tar.gz"
+checksum=0386336681b24d648cdce3daa0c2b83c77a69773296e97a8427b67cbd27531ed
+
```
Comment (Hoshpak) [inline srcpkgs/ddgtk/template]: please remove this empty line.

Comment (reback00) [reply srcpkgs/ddgtk/template]: done.

**Pull request merged**